diff --git a/CMakeLists.txt b/CMakeLists.txt
index 94c0bfb..ba8ab03 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -502,7 +502,6 @@ include_directories( ${THIRDPARTYDIR}/include )
 
 ### Handle third-party CEREAL
 include_directories( ${THIRDPARTYDIR}/cereal/include )
-install(DIRECTORY ${THIRDPARTYDIR}/cereal/include/ DESTINATION include/palisade)
 
 include_directories( ${CMAKE_SOURCE_DIR}/third-party/google-test/googletest )
 include_directories( ${CMAKE_SOURCE_DIR}/third-party/google-test/googletest/include )
